scripts/build/types.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package build
+
+import (
+	"os"
+)
+
+// Config contains various pieces of configuration created by main and passed
+// around.
+type Config struct {
+	ArchiveFormat string
+	BuildFormats  []string
+	OutputPath    string
+	Packages      []string
+	Run           bool
+	Verbose       bool
+}
+
+// Intermediate files are stored in memory so you do not need sudo to create
+// cpio files. Some fields such as atime and mtime are not included here
+// because they are useless for our purposes.
+type file struct {
```

Comment:
I wasn't thinking straight -- in Go, the most idiomatic solution is an io.Reader.